src/item.cpp

```diff
@@ -4237,134 +4237,55 @@ void item::detonate(point p) const
     g->explosion(p.x, p.y, type->explosion_on_fire_data.power, type->explosion_on_fire_data.shrapnel, type->explosion_on_fire_data.fire, type->explosion_on_fire_data.blast);
 }
 
-
-//sort quivers by contents, such that empty quivers go last
-struct sort_by_charges {
-    bool operator()(const std::pair<item*,int> &left, const std::pair<item*,int> &right) {
-        if(left.first->contents.empty()) {
-            return false;
-        } else if (right.first->contents.empty()){
-            return true;
-        } else {
-            return right.first->contents[0].charges < left.first->contents[0].charges;
-        }
+bool item_compare_by_charges( const item *left, const item *right)
```

Comment:
Compare functions traditionally return an int, either positive, negative or 0 to avoid the problem where you can't tell a == b apart from a > b.  You can always just test for < 0 if that's what you care about, but it allows detecting the == case as well.